### ISSUE TITLE
Separate static symbols from global in C, C++

### ISSUE THREAD

**Issue by makelinux** [doxygen/doxygen#9393]
### Current view:

Mixed globals and statics 

![Screenshot from 2022-06-11 20-03-59](https://user-images.githubusercontent.com/2335185/173197586-dbb8ec41-3f42-4281-9e86-5d46e55bdc71.png)

### Proposed view:

**Globals**

```
void  readConfiguration (int argc, char **argv)
void  checkConfiguration ()
void  adjustConfiguration ()
void  searchInputFiles ()
void  parseInput ()
void  generateOutput ()

```
**Statics**

```
static void  stopDoxygen (int)
static void  writeTagFile ()
static void  exitDoxygen ()
static QCString createOutputDirectory (const QCString &baseDirName, const QCString &formatDirName, const char *defaultDirName)
```

**Comment by albert-github** [comment]
Interesting but difficult problem as the "Functions" are not only used in the list of brief descriptions as shown, but also on a number of different places a.o.:
- in the layout file and through this it is used automatically on a number of places.
- in the top right corner of the screen:
   ![image](https://user-images.githubusercontent.com/5223533/173225878-c33787cc-f91f-493f-bb58-3e5a8d24a45e.png)

A thing that I could image is to split the brief description into multiple categories from:
![image](https://user-images.githubusercontent.com/5223533/173226018-e229f8bd-07e3-4cc9-b178-c4fc88766f84.png)

to:
![image](https://user-images.githubusercontent.com/5223533/173225997-1685bf36-64b9-4df5-ad32-47fcf843a2ea.png)

Note: this will not only be valid for C / C++ but for all global / static functions

In case we have e.g. no global functions I could imagine that we get:
![image](https://user-images.githubusercontent.com/5223533/173226177-91bff472-2d1c-4496-ba2f-8ef5fb0d1cea.png)



@makelinux, @doxygen what do you think?

**Comment by makelinux** [comment]
Please don't forget about static variables. Also rending of statics is optional.

**Comment by makelinux** [comment]
Statics are like private members.
I propose to group statics after globals:

 ## Functions
 
 ## Variables
 
 ## Static functions
 
 ## Static variables

**Comment by doxygen** [comment]
I agree, it makes sense to separate the public API from the private one. if you use member grouping this is already possible today:

```c
/** @file */

/** @name Globals
 *  @{
 */
void  readConfiguration(int argc, char **argv) {}
void  checkConfiguration() {}
void  adjustConfiguration() {}
void  searchInputFiles() {}
void  parseInput() {}
void  generateOutput() {}
/** @} */

/** @name Statics
 *  @{
 */
static void  stopDoxygen(int) {}
static void  writeTagFile() {}
static void  exitDoxygen() {}
/** @} */
```

So this sounds like request to automate this grouping, which should then not interfere with the member groups a user may  already have used. 

Note that in C++ an anonymous namespaces can be used as an alternative to `static` and in C++20 modules are introduced providing yet another way to split public and private information.
